app.py

```diff
@@ -106,6 +117,10 @@ def app(
         default=LogLevel.INFO,
         help="Verbosity of log. Only 'debug' or 'info' is relevant.",
     ),
+    strict_process: bool = typer.Option(
+        default=False,
+        help="If set, only re-encrypt the exact files which were wrongly encrypted.",
+    ),
```

Comment:
There is an alternative, instead of having a `strict_process` bool flag, it is also possible to slip in a new `Match.STRICT`.

It would make sense since choosing it excludes all other `Match` type, an advantage over using `strict_process` where it takes manual explanation that using this flag makes the `Match` flag pointless.

However it won't have a corresponding `pred` to map into + uses a different logic from the other `Match`. This is where now `strict_process` has an advantage over this since it's logically a different flag to begin with and therefore intuitive to expect a different logic from the filter pred + you can put in special message to describe it more.

Either way is fine, maybe for `strict_process` (the current method), perhaps make it clear in the help message that it nullifies the use of `filter_match`?